packages/lesswrong/server/scripts/mergeAccounts.js

```diff
@@ -0,0 +1,149 @@
+      })
+    }
+  }
+  documents.forEach((doc) => {
+    
+  })
+}
+
+const transferEditableField = ({documentId, targetUserId, collection, fieldName = "contents"}) => {
+  // Update the denormalized revision on the document
+  editMutation({
+    collection,
+    documentId,
+    set: {[`${fieldName}.userId`]: targetUserId},
+    unset: {},
+    validate: false
+  })
+  // Update the revisions themselves
+  Revisions.update({ documentId, fieldName }, {$set: {userId: targetUserId}}, { multi: true })
+}
+
+const mergeReadStatusForPost = ({sourceUserId, targetUserId, postId}) => {
+  const sourceUserStatus = ReadStatuses.findOne({userId: sourceUserId, postId})
+  const targetUserStatus = ReadStatuses.findOne({userId: targetUserId, postId})
+  const readStatus = !!(sourceUserStatus?.isRead || targetUserStatus?.isRead)
+  const lastUpdated = new Date(sourceUserStatus?.lastUpdated) > new Date(targetUserStatus?.lastUpdated) ? sourceUserStatus?.lastUpdated : targetUserStatus?.lastUpdated
+  if (targetUserStatus) {
+    ReadStatuses.update({_id: targetUserStatus._id}, {$set: {isRead: readStatus, lastUpdated}})
+  } else if (sourceUserStatus) {
+    // eslint-disable-next-line no-unused-vars
+    const {_id, ...sourceUserStatusWithoutId} = sourceUserStatus
+    ReadStatuses.insert({...sourceUserStatusWithoutId, userId: targetUserId})
+  }
+}
+
+Vulcan.mergeAccounts = async (sourceUserId, targetUserId) => {
+  const sourceUser = Users.findOne({_id: sourceUserId})
+  const targetUser = Users.findOne({_id: targetUserId})
+
+  // Transfer posts
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Posts"})
+
+  // Transfer comments
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Comments"})
+
+  // Transfer conversations
+  await Conversations.update({participantIds: sourceUserId}, {$set: {"participantIds.$": targetUserId}}, { multi: true })
+
+  // Transfer private messages
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Messages"})
+
+  // Transfer notifications
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Notifications"})
+
+  // Transfer readStatuses
+  const readStatuses = await ReadStatuses.find({userId: sourceUserId}).fetch()
+  const readPostIds = readStatuses.map((status) => status.postId)
+  readPostIds.forEach((postId) => {
+    mergeReadStatusForPost({sourceUserId, targetUserId, postId})
+  })
+
+  // Transfer sequences
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Sequences"})
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Collections"})
+  
+  // Transfer karma
+  // eslint-disable-next-line no-console
+  console.log("Transferring karma")
+  await editMutation({
+    collection: Users,
+    documentId: targetUserId,
+    set: {karma: sourceUser.karma + targetUser.karma, afKarma: sourceUser.afKarma + targetUser.afKarma},
```

Comment:
Do we have a "recompute karma from scratch" utility function? That would work better (and handle corner cases like if you merge two accounts which have voted on each other.)

Reply:
Maybe, I was thinking of the same, but couldn't find one super easily